chromadb/utils/embedding_functions.py

```diff
@@ -30,7 +30,81 @@
 
 logger = logging.getLogger(__name__)
 
+if os.getenv('GENERAL_SENTENCES') == "True":
+    from AutoModelForSentenceEmbedding import (
+        AutoModelForSentenceEmbedding,
+    )
+else:
+    print("Environment variable is not set, skipping import.")
 
+class JinaWithPeftEmbeddingsFunction(EmbeddingFunction[Documents]):
+    # Since we do dynamic imports we have to type this as Any
+    models: Dict[str, Any] = {}
+    def __init__(
+        self,
+        # model_name: str = "jinaai/jina-embeddings-v2-base-en", # TODO make SentenceTransformer branch with PEFT
+        adapters_path: str = "",
+        device: str = "cpu",
+        normalize_embeddings: bool = True,
+    ):
+        model_name = "jinaai/jina-embeddings-v2-base-en"
+        if model_name not in self.models:
+            try:
+                from transformers import AutoTokenizer
+            except ModuleNotFoundError:
+                print("transformers module is not installed. You can install it by running 'pip install transformers' in your application runtime terminal.")
+            tokenizer = AutoTokenizer.from_pretrained(
+                model_name,
+            )    
+            model = AutoModelForSentenceEmbedding(model_name, tokenizer)
+            if len(adapters_path) > 0:
+                self._peft = True
+                print("BASE", model)
+                
+                try:
+                    from peft import PeftModel
+                except ModuleNotFoundError:
+                    print("peft module is not installed. You can install it by running 'pip install peft' in your application runtime terminal.")
+                
+                model = PeftModel.from_pretrained(model, adapters_path)
+                model.to(device)
+                model.eval()
+                model = model.merge_and_unload()
+                
+            else:
+                self._peft = False
+            self.models[model_name] = model
+        self._model_name = model_name
+        self._model = self.models[model_name]
+        self._device = device
+        self._tokenizer = tokenizer
+        self._normalize_embeddings = normalize_embeddings
+    def __call__(self, input: Documents) -> Embeddings:
+        device = self._device
+        if len(input) <= 0:
+            return []
+        if self._peft:
```

Comment:
nit: this code would be easier to read if we swapped the conditional branches.

```python
if not self._peft:
    return self._model.encode(
        list(input),
        convert_to_numpy=True,
        normalize_embeddings=self._normalize_embeddings,
    ).tolist()

embeddings = []
try:
    from torch import inference_mode
...
```

Moving the short `else` branch into the primary `if`-statement lets us remove a layer of indentation from the rest of the code.